scripts/e2e-tests.sh

```diff
@@ -323,8 +323,8 @@ function setup_test_cluster() {
     abort "kubeconfig context set to ${k8s_cluster}, which is forbidden"
 
   # If cluster admin role isn't set, this is a brand new cluster
-  # Setup the admin role and also KO_DOCKER_REPO
-  if [[ -z "$(kubectl get clusterrolebinding cluster-admin-binding 2> /dev/null)" ]]; then
+  # Setup the admin role and also KO_DOCKER_REPO if it is a GKE cluster
+  if [[ -z "$(kubectl get clusterrolebinding cluster-admin-binding 2> /dev/null)" && "${k8s_cluster}" =~ ^gke_.* ]]; then
     acquire_cluster_admin_role ${k8s_user} ${E2E_CLUSTER_NAME} ${E2E_CLUSTER_REGION} ${E2E_CLUSTER_ZONE}
     kubectl config set-context ${k8s_cluster} --namespace=default
```

Comment:
would this be needed for other providers?

Reply:
Makes sense. And KO_DOCKER_REPO should be set outside this `if` as well.

Reply:
yeah, but then we'll need to discuss what the docker registry should be since user might not have gcloud access

Reply:
On a second though, nevermind moving KO_DOCKER_REPO. Already existing GCP environments and non-GCP environment will have to set it anyway.

Reply:
I'm even not sure if `kubectl config set-context ${k8s_cluster} --namespace=default` is needed for `gcloud`. In the previous logic, this line will only be executed on a brand new cluster, which must already have namespace set to `default`.
But I think it makes sense to use the `default` namespace  as the default one for the current context in e2e tests, so I moved it outside the `if`.